external/llvm-project/mlir/lib/Transforms/Utils/OutlinerUtils.cpp

```diff
@@ -173,337 +209,380 @@ static bool opsMatch(Operation *lhs, Operation *rhs, OutliningCandidate &one,
 
     // Otherwise, these operands must have the same logical order within the
     // parent block.
-    if (one.getOrderOf(lhsOperand) != two.getOrderOf(rhsOperand)) {
+    if (getOrderOf(lhsOperand) != two.getOrderOf(rhsOperand)) {
       return false;
     }
   }
 
   return true;
 }
 
-static bool outliningCandidatesEquivalent(OutliningCandidate &one,
-                                          OutliningCandidate &two) {
-  if (one.hash != two.hash) {
-    return false;
+bool OutliningCandidate::equivalent(const OutliningCandidate &that) const {
+  if (hash == that.hash) {
+    if (params.size() != that.params.size() || ops.size() != that.ops.size()) {
+      return false;
+    }
+    for (auto params : llvm::zip(params, that.params)) {
+      if (std::get<0>(params) != std::get<1>(params)) {
+        return false;
+      }
+    }
+    // get Result Types
+
+    for (auto opPair : llvm::zip(ops, that.ops)) {
+      if (!opsMatch(std::get<0>(opPair), std::get<1>(opPair), that)) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
Could this be the built-in `OperationEquivalence` with appropriate flags set?

Reply:
It started that way, but I didn't note why I made my own version.  Could be it didn't quite do what I wanted, or could be that I didn't want to mess with mainstream code (back in 2021).